rfc/src/rfcs/0004-loop-contract-synthesis.md

```diff
@@ -0,0 +1,145 @@
+- **Feature Name:** Loop-contract synthesis
+- **Feature Request Issue:** <https://github.com/model-checking/kani/issues/2214>
+- **RFC PR:** <https://github.com/model-checking/kani/pull/2215>
+- **Status:** Under Review
+- **Version:** 0
+- **Proof-of-concept:** <https://github.com/qinheping/kani/tree/kani-synthesizer>
+
+## Summary
+
+An new option that allows users to verify programs without unwinding loops by synthesizing loop contracts for those loops.
+
+
+## User Impact
+Currently Kani does not support proving correctness (i.e. assertions never fail) on programs with unbounded control flow (e.g. loops with dynamic bounds). Kani unrolls all unbounded loops until a few iterations and then verifies this unrolled program — it thus provides a much weaker guarantee on correctness.
```

Comment:
I would re-phrase "it thus provides a much weaker guarantee on correctness." to see bounds as part of the assumptions.